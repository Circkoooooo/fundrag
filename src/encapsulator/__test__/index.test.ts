import React from 'react'
import { createDragComponent } from '..'

describe('encapsulate react component to drag component', () => {
	it('encapsulate a instance', () => {
		const realNode = () => {
			return React.createElement('div', {}, '123')
		}
		let dragComponent = createDragComponent(realNode, {
			isContainer: false,
			componentName: '线性组件',
		})
		expect(dragComponent).toMatchSnapshot()
		dragComponent = createDragComponent(realNode, {
			isContainer: true,
			componentName: '线性组件',
		})
		expect(dragComponent).toMatchSnapshot()
	})
})
