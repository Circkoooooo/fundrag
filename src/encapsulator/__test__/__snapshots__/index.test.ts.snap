// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encapsulate react component to drag component encapsulate a instance 1`] = `
Object {
  "RealNode": [Function],
  "children": Object {
    "appendChild": [Function],
    "value": Array [],
  },
  "componentRenderConfig": Object {
    "componentName": "线性组件",
  },
  "isContainer": false,
}
`;

exports[`encapsulate react component to drag component encapsulate a instance 2`] = `
Object {
  "RealNode": [Function],
  "children": Object {
    "appendChild": [Function],
    "value": Array [],
  },
  "componentRenderConfig": Object {
    "componentName": "线性组件",
  },
  "isContainer": true,
}
`;
